[Inspector V2] Request the reduced widget tree from Flutter

elliette opened the pull request:
Work towards https://github.com/flutter/devtools/issues/1423

Uses the `fullDetails=false` parameter added in https://github.com/flutter/flutter/pull/157309 to request the smaller widget tree.

kenzieschmoll reviewed (commented):
Is there a way to test this to verify we are getting the smaller tree? Perhaps an inspector text golden test that checks this case?

elliette commented:
> Is there a way to test this to verify we are getting the smaller tree? Perhaps an inspector text golden test that checks this case?

Good point, added a test case to the integration test (which is using a real app). I've added a check that the JSON object for each node only contains the information we expect for the small tree.

elliette commented on packages/devtools_app/test/inspector_v2/inspector_integration_test.dart:
```diff
@@ -289,6 +289,47 @@ void main() {
+      final rowsInTree = state.controller.inspectorTree.rowsInTree.value;
+
+      for (final row in rowsInTree) {
+        final detailKeys = row?.node.diagnostic?.json.keys ?? [];
```
ah good point, done!

bkonyi commented on packages/devtools_app/test/inspector_v2/inspector_integration_test.dart:
```diff
@@ -289,6 +290,59 @@ void main() {
+      final rowsInTree = state.controller.inspectorTree.rowsInTree.value;
+
+      for (final row in rowsInTree) {
+        final detailKeys = row?.node.diagnostic?.json.keys ?? [];
```
Ubernit: the default list should be `const <String>[]`

elliette replied:
done!

bkonyi reviewed (approved):
LGTM overall!
